src/AutoRest.CSharp/LowLevel/Generation/LowLevelClientWriter.cs

```diff
@@ -89,19 +87,69 @@ private void WriteClientMethod(CodeWriter writer, RestClientMethod clientMethod,
                 writer.Append($");");
                 writer.Line();
 
+                using (writer.Scope($"if (requestOptions.PerCallPolicy != null)"))
+                {
+                    writer.Line($"message.SetProperty (\"RequestOptionsPerCallPolicyCallback\", requestOptions.PerCallPolicy);");
+                }
+
                 if (async)
                 {
-                    writer.Line($"return await {PipelineField:I}.SendRequestAsync(req, cancellationToken).ConfigureAwait(false);");
+                    writer.Line($"await {PipelineField:I}.SendAsync(message, requestOptions.CancellationToken).ConfigureAwait(false);");
                 }
                 else
                 {
-                    writer.Line($"return {PipelineField:I}.SendRequest(req, cancellationToken);");
+                    writer.Line($"{PipelineField:I}.Send(message, requestOptions?.CancellationToken ?? System.Threading.CancellationToken.None);");
+                }
+
+                writer.Line($"ResponseStatusOption statusOption = requestOptions?.StatusOption ?? ResponseStatusOption.Default;");
```

Comment:
Do we need the `?.` given that we ensure this is not null on L78?

Reply:
(Similar feedback for the use of cacellationtoken)

Reply:
(FWIW, I fixed this when I started adding commits, but I'm keeping the issue open for you to ack)